File: pkg/deltatimer/deltatimer.go
package deltatimer

import "time"

type DeltaTimer struct {
	lastTick int64
}

func New() *DeltaTimer {
	return &DeltaTimer{
		lastTick: time.Now().UnixMicro(),
	}
}

func (timer *DeltaTimer) GetDeltaTimeMs() float64 {
	now := time.Now().UnixMicro()
	dt := now - timer.lastTick
	timer.lastTick = now

	return float64(dt) / 1000.0
}

File: web/typescript/gameassets/room/deltaTimer.ts
class DeltaTimer {
    private lastTick: DOMHighResTimeStamp;

    constructor(){
        this.lastTick = performance.now();
    }

    getDeltaTime(){
        let now = performance.now();
        let dt = now - this.lastTick;
        this.lastTick = now;

        return dt;
    }
}